Fix pos_constr_comma Trail

guillaume-roche opened the pull request:
**Reopening a PR for merge on `genymobile` instead of `master`.**

When `pos_constr_comma` is set to `Trail`, leading commas are not properly handled: preceding new lines is not removed, but a following newline is added. This results in having commas in their own lines.

Input:

```
MyClass::MyClass(Type *var1, Type *var2) :
    MySuperClass(parent)
  , mVar1(var1)
  , mVar2(var2) {
}
```

Output:

```
MyClass::MyClass(Type *var1, Type *var2) :
    MySuperClass(parent)
    ,
    mVar1(var1)
    ,
    mVar2(var2) {
}
```

Expected:

```
MyClass::MyClass(Type *var1, Type *var2) :
    MySuperClass(parent),
    mVar1(var1),
    mVar2(var2) {
}
```

This patch fixes this behaviour.

agateau-g commented:
Can you extend the existing unit-tests to catch this? Will make it easier to upstream.

agateau-g committed:
Fix pos_constr_comma Trail

When `pos_constr_comma` is set to `Trail`, leading commas are not
properly handled: preceding new line is not removed, but a following
newline is added. This results in having commas in their own lines.

This patch fixes this behavior.
